src/components/Swap/Toolbar/Caption.tsx

```diff
@@ -68,57 +89,62 @@ export function MissingInputs() {
   return <Caption icon={Info} caption={<Trans>Enter an amount</Trans>} />
 }
 
-export function LoadingTrade() {
+export function LoadingTrade({ gasUseEstimateUSD }: GasEstimateProp) {
   return (
-    <Caption
-      icon={InlineSpinner}
-      caption={
-        <Loading>
-          <Trans>Fetching best price…</Trans>
-        </Loading>
-      }
-    />
+    <>
+      <Caption
+        icon={InlineSpinner}
+        caption={
+          <Loading>
+            <Trans>Fetching best price…</Trans>
+          </Loading>
+        }
+      />
+      <GasEstimate gasUseEstimateUSD={gasUseEstimateUSD} />
+    </>
   )
 }
 
-export function WrapCurrency({ inputCurrency, outputCurrency }: { inputCurrency: Currency; outputCurrency: Currency }) {
+interface WrapCurrencyProps {
```

Comment:
nit: Pulling this and TradeProps into separate interfaces is a choice of personal preference - I'm fine with it either way. They are not heavily used interfaces - they are both one-offs.